### ISSUE TITLE
test_deprecate fails on Windows Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8756]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8756|
|Type|enhancement|
|Created|2016-08-14 09:37:28Z|
|Branch|https://github.com/twisted/twisted/tree/8756-rodrigc-deprecate-py3|

**twisted.python.test.test_deprecate** fails on Windows with Python 3.5

```
[FAIL]
Traceback (most recent call last):
  File "c:\projects\twisted\build\py35-coverage-windows\lib\site-packages\twisted\python\test\test_deprecate.py", line 522, in test_filteredOnceWarning
    self.assertEqual(len(warningsShown), 1)
  File "c:\projects\twisted\build\py35-coverage-windows\lib\site-packages\twisted\trial\_synctest.py", line 425, in assertEqual
    super(_Assertions, self).assertEqual(first, second, msg)
  File "C:\PYTHON35-x64\Lib\unittest\case.py", line 820, in assertEqual
    assertion_func(first, second, msg=msg)
  File "C:\PYTHON35-x64\Lib\unittest\case.py", line 813, in _baseAssertEqual
    raise self.failureException(msg)
twisted.trial.unittest.FailTest: 29 != 1
twisted.python.test.test_deprecate.WarnAboutFunctionTests.test_filteredOnceWarning
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "c:\projects\twisted\build\py35-coverage-windows\lib\site-packages\twisted\python\test\test_deprecate.py", line 500, in test_filteredWarning
    self.assertEqual(len(warningsShown), 0)
  File "c:\projects\twisted\build\py35-coverage-windows\lib\site-packages\twisted\trial\_synctest.py", line 425, in assertEqual
    super(_Assertions, self).assertEqual(first, second, msg)
  File "C:\PYTHON35-x64\Lib\unittest\case.py", line 820, in assertEqual
    assertion_func(first, second, msg=msg)
  File "C:\PYTHON35-x64\Lib\unittest\case.py", line 813, in _baseAssertEqual
    raise self.failureException(msg)
twisted.trial.unittest.FailTest: 28 != 0
twisted.python.test.test_deprecate.WarnAboutFunctionTests.test_filteredWarning
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "c:\projects\twisted\build\py35-coverage-windows\lib\site-packages\twisted\python\test\test_deprecate.py", line 476, in test_renamedFile
    self.assertSamePath(warnedPath, expectedPath)
  File "c:\projects\twisted\build\py35-coverage-windows\lib\site-packages\twisted\python\test\test_deprecate.py", line 443, in assertSamePath
    "%r != %r" % (first, second))
  File "c:\projects\twisted\build\py35-coverage-windows\lib\site-packages\twisted\trial\_synctest.py", line 388, in assertTrue
    super(_Assertions, self).assertTrue(condition, msg)
  File "C:\PYTHON35-x64\Lib\unittest\case.py", line 677, in assertTrue
    raise self.failureException(msg)
twisted.trial.unittest.FailTest: False is not true : FilePath(b'c:\\projects\\twisted\\build\\py35-coverage-windows\\lib\\ntpath.py') != FilePath(b'C:\\projects\\twisted\\build\\py35-coverage-windows\\tmp\\_trial_temp\\twisted.python.test.test_depreca\\WarnAboutFunctionTests\\test_renamedFile\\0c9xqex0\\temp\\twisted_renamed_helper\\module.py')
twisted.python.test.test_deprecate.WarnAboutFunctionTests.test_renamedFile
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "c:\projects\twisted\build\py35-coverage-windows\lib\site-packages\twisted\python\test\test_deprecate.py", line 409, in test_warning
    FilePath(warningsShown[0]["filename"]), FilePath(filename))
  File "c:\projects\twisted\build\py35-coverage-windows\lib\site-packages\twisted\python\test\test_deprecate.py", line 443, in assertSamePath
    "%r != %r" % (first, second))
  File "c:\projects\twisted\build\py35-coverage-windows\lib\site-packages\twisted\trial\_synctest.py", line 388, in assertTrue
    super(_Assertions, self).assertTrue(condition, msg)
  File "C:\PYTHON35-x64\Lib\unittest\case.py", line 677, in assertTrue
    raise self.failureException(msg)
twisted.trial.unittest.FailTest: False is not true : FilePath('c:\\projects\\twisted\\build\\py35-coverage-windows\\lib\\ntpath.py') != FilePath('c:\\projects\\twisted\\build\\py35-coverage-windows\\lib\\site-packages\\twisted\\python\\test\\test_deprecate.py')
twisted.python.test.test_deprecate.WarnAboutFunctionTests.test_warning
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "c:\projects\twisted\build\py35-coverage-windows\lib\site-packages\twisted\python\test\test_deprecate.py", line 423, in test_warningLineNumber
    self.package.sibling(b'twisted_private_helper').child(b'module.py'))
  File "c:\projects\twisted\build\py35-coverage-windows\lib\site-packages\twisted\python\test\test_deprecate.py", line 443, in assertSamePath
    "%r != %r" % (first, second))
  File "c:\projects\twisted\build\py35-coverage-windows\lib\site-packages\twisted\trial\_synctest.py", line 388, in assertTrue
    super(_Assertions, self).assertTrue(condition, msg)
  File "C:\PYTHON35-x64\Lib\unittest\case.py", line 677, in assertTrue
    raise self.failureException(msg)
twisted.trial.unittest.FailTest: False is not true : FilePath(b'c:\\projects\\twisted\\build\\py35-coverage-windows\\lib\\ntpath.py') != FilePath(b'C:\\projects\\twisted\\build\\py35-coverage-windows\\tmp\\_trial_temp\\twisted.python.test.test_depreca\\WarnAboutFunctionTests\\test_warningLineNumber\\hmul5vw4\\temp\\twisted_private_helper\\module.py')
twisted.python.test.test_deprecate.WarnAboutFunctionTests.test_warningLineNumber
===============================================================================
```

<details><summary>Searchable metadata</summary>

```
trac-id__8756 8756
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8756_rodrigc_deprecate_py3 8756-rodrigc-deprecate-py3
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1471167448661169 1471167448661169
changetime__1471724876917302 1471724876917302
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_2"></a>|@rodrigc commented|
|-|-|

https://github.com/twisted/twisted/pull/495

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc set owner to @rodrigc|
|-|-|

Reviewed by itamar https://github.com/twisted/twisted/pull/495#issuecomment-241219289

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_4"></a>|@rodrigc set status to `closed`|
|-|-|

In changeset 1aaf33f2210ae514e6b0c7e3b7930446173c3a32
```
#!CommitTicketReference repository="" revision="1aaf33f2210ae514e6b0c7e3b7930446173c3a32"
Merge pull request #495 from twisted/8756-rodrigc-deprecate-py3

Author: hawkowl, rodrigc
Reviewer: itamar
Fixes: #8756

Fix test_deprecate.py in Windows with Python 3
```
